workspaces/mcp-chat/plugins/mcp-chat-backend/src/providers/claude-provider.ts

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  LLMProvider,
+  ChatMessage,
+  Tool,
+  ChatResponse,
+  ToolCall,
+} from './base-provider';
+
+export class ClaudeProvider extends LLMProvider {
+  async sendMessage(
+    messages: ChatMessage[],
+    tools?: Tool[],
+  ): Promise<ChatResponse> {
+    const requestBody = this.formatRequest(messages, tools);
+    const response = await this.makeRequest('/messages', requestBody);
+    return this.parseResponse(response);
+  }
+
+  async testConnection(): Promise<{
+    connected: boolean;
+    models?: string[];
+    error?: string;
+  }> {
+    try {
+      // Claude doesn't have a models endpoint, so we'll make a simple test request
+      const testMessages = [{ role: 'user' as const, content: 'Hello' }];
+      const requestBody = {
+        model: this.model,
+        max_tokens: 1,
+        messages: this.convertToAnthropicFormat(testMessages),
+      };
+
+      const response = await fetch(`${this.baseUrl}/messages`, {
+        method: 'POST',
+        headers: this.getHeaders(),
+        body: JSON.stringify(requestBody),
+      });
+
+      if (!response.ok) {
+        const errorText = await response.text();
+        let errorMessage = `Claude API error (${response.status})`;
+
+        try {
+          const errorData = JSON.parse(errorText);
+          if (errorData.error?.message) {
+            errorMessage = errorData.error.message;
+          }
+        } catch {
+          // If parsing fails, use the raw error text but truncate if too long
+          errorMessage =
+            errorText.length > 100
+              ? `${errorText.substring(0, 100)}...`
+              : errorText;
+        }
+
+        // Provide user-friendly messages for common errors
+        if (response.status === 401) {
+          errorMessage =
+            'Invalid API key. Please check your Claude API key configuration.';
+        } else if (response.status === 429) {
+          errorMessage =
+            'Rate limit exceeded. Please try again later or check your Claude usage limits.';
+        } else if (response.status === 403) {
+          errorMessage =
+            'Access forbidden. Please check your API key permissions.';
+        }
+
+        return {
+          connected: false,
+          error: errorMessage,
+        };
```

Comment:
Not critical, but maybe an idea for the future. If you return the status code and origin message to the frontend, couy can truncate and translate these strings there easier.

Not necessary to me to change this right now.